### PR TITLE
Develop --> Master

### PR DESCRIPTION
**Pull request by hasueki**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -46,9 +47,11 @@ require (
...
-	k8s.io/api v0.21.0
-	k8s.io/apimachinery v0.21.0
+	k8s.io/api v0.21.1
+	k8s.io/apimachinery v0.21.1
 	k8s.io/client-go v0.21.0
 	k8s.io/klog/v2 v2.8.0
 	k8s.io/utils v0.0.0-20210305010621-2afb4311ab10
 )
+
+replace github.com/openshift/api => github.com/hasueki/api v0.0.0-20210616144002-767ee5979974
```
Comment (srcarrier) [inline go.mod]: Is this `replace` statement just for testing/development _(i.e. to be removed in the upstream PR)_?

Comment (hasueki) [reply go.mod]: Yes, that's correct. Will remove once the `openshift/api` changes land in the upstream repo.

**Comment by srcarrier** [comment]
The other providers include test files `/pkg/storage/<PROVIDER>/*_test.go` alongside their storage implementations.
No harm in opening a PR, but they might want some test functions to go with it, unless there's a yet-to-be-met dependency that precludes us from doing so at this point in time.

**Comment by hasueki** [comment]
@srcarrier This PR isn't actually pointing to the upstream repo, `openshift/cluster-image-registry-operator`. This is going into the master branch in my forked repo. I opened this PR just so it'll be easier to view the diff and review. I am starting to look at the unit tests now.

File: pkg/storage/ibmcos/ibmcos.go [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+	"github.com/openshift/cluster-image-registry-operator/pkg/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/envvar"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
+)
+
+const (
+	IAMEndpoint                   = "https://iam.cloud.ibm.com/identity/token"
```
Comment (srcarrier) [inline pkg/storage/ibmcos/ibmcos.go:34]: I do wonder if hard-coding some of these endpoints will come back to haunt us in the future.
If hard-coding is a must, is there some way to define the endpoint in fewer places or is there a way to override it?
Seems like this precludes us from running this in staging against the `https://iam.test.cloud...` instance.

**Pull request merged**